### PR TITLE
Update upstream 3443f154

### LINKED ISSUE #1967: update to upstream worker commit: 3443f154
### Context
As title, there are some difference between our local copy and upstream worker.

### Task
Sync and fill the gap.

### Acceptance criteria
What's the definition of done, if applicable?

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by BillyWooo**
### Context

As title, update to upstream [worker](https://github.com/integritee-network/worker) commit 3443f154

resolves #1967

### CODE REVIEW

File: tee-worker/.github/workflows/build_and_test.yml
Comment (Kailai-Wang) [inline tee-worker/.github/workflows/build_and_test.yml:1]: Do we need to any some of the changes to our `ci.yml`?

File: tee-worker/Makefile [hunk 1]
```diff
@@ -75,11 +75,13 @@ ifeq ($(SGX_PRODUCTION), 1)
 	SGX_ENCLAVE_MODE = "Production Mode"
 	SGX_ENCLAVE_CONFIG = "enclave-runtime/Enclave.config.production.xml"
 	SGX_SIGN_KEY = $(SGX_COMMERCIAL_KEY)
+	SGX_SIGN_PASSFILE = $(SGX_PASSFILE)
```
Comment (Kailai-Wang) [inline tee-worker/Makefile:78]: In production how can we get the `SGX_PASSFILE`? Why do we need it now (not before)?

**Review by Kailai-Wang — APPROVED** [review]
Looks fine in general.

I left some questions - we need to figure them out sooner or later.
We also need to start to use `data-dir` command

File: tee-worker/.github/workflows/build_and_test.yml
Comment (BillyWooo) [reply tee-worker/.github/workflows/build_and_test.yml:1]: There are some interesting changes:
- changes related to IAS and dcap -> not applicable for the moment
- changes about rm used docker images -> let me do it in another small PR.

File: tee-worker/Makefile (hunk 1 above)
Comment (BillyWooo) [reply tee-worker/Makefile:78]: Discuss in private channel.

**Comment by BillyWooo** [comment]
Path of `GLOBAL_SCHEDULED_ENCLAVE` related issue deserve another small issue/PR.

**Pull request merged**